junit-jupiter-params/src/main/java/org/junit/jupiter/params/ParameterizedTestExtension.java

```diff
@@ -82,6 +90,7 @@ public Stream<TestTemplateInvocationContext> provideTestTemplateInvocationContex
 				.map(provider -> AnnotationConsumerInitializer.initialize(methodContext.method, provider))
 				.flatMap(provider -> arguments(provider, extensionContext))
 				.map(arguments -> {
+					validateArgumentCount(extensionContext, arguments);
 					invocationCount.incrementAndGet();
```

Comment:
I think we need to flip these two lines to avoid `onClose()` below throwing another exception that will be reported as suppressed:

```
Failures (1):
  JUnit Jupiter:ParameterizedTestDemo:testWithArgumentCountValidation(int)
    MethodSource [className = 'example.ParameterizedTestDemo', methodName = 'testWithArgumentCountValidation', methodParameterTypes = 'int']
    => org.junit.platform.commons.PreconditionViolationException: Configuration error: the @ParameterizedTest has 1 argument(s) but there were 2 argument(s) provided.
Note: the provided arguments are [42, -666]
       java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
       java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:184)
       java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1708)
       java.base/java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:762)
       java.base/java.util.stream.ReferencePipeline$7$1.accept(ReferencePipeline.java:276)
       [...]
       Suppressed: org.junit.platform.commons.PreconditionViolationException: Configuration error: You must configure at least one set of arguments for this @ParameterizedTest
         java.base/java.util.stream.AbstractPipeline.close(AbstractPipeline.java:323)
         [...]
```